src/Selection.js

```diff
@@ -113,92 +191,109 @@ class Selection {
         bottom: offsetData.bottom + bottom,
         right: offsetData.right + right
       },
-      { top: e.pageY, left: e.pageX });
+      { top: pageY, left: pageX });
 
       if (!collides) return;
     }
 
-    let result = this.emit('mousedown', this._mouseDownData = {
-      x: e.pageX,
-      y: e.pageY,
-      clientX: e.clientX,
-      clientY: e.clientY
+    let result = this.emit('beforeSelect', this._initialEventData = {
+      isTouch: /^touch/.test(e.type),
+      x: pageX,
+      y: pageY,
+      clientX,
+      clientY,
     });
 
     if (result === false)
       return;
 
     //e.preventDefault();
 
-    this._onMouseUpListener = addEventListener('mouseup', this._mouseUp)
-    this._onMouseMoveListener = addEventListener('mousemove', this._openSelector)
+    switch (e.type) {
+      case 'mousedown':
+        this._onEndListener = addEventListener('mouseup', this._handleTerminatingEvent)
+        this._onMoveListener = addEventListener('mousemove', this._handleMoveEvent)
+        break;
+      case 'touchstart':
+        this._handleMoveEvent(e);
+        this._onEndListener = addEventListener('touchend', this._handleTerminatingEvent)
+        this._onMoveListener = addEventListener('touchmove', this._handleMoveEvent)
+        break;
+      default:
+        break;
+    }
   }
 
-  _mouseUp(e) {
+  _handleTerminatingEvent(e) {
+    const { pageX, pageY, clientX, clientY } = getEventCoordinates(e);
+
+    this.selecting = false;
 
-    this._onMouseUpListener && this._onMouseUpListener.remove();
-    this._onMouseMoveListener && this._onMouseMoveListener.remove();
+    this._onEndListener && this._onEndListener.remove();
+    this._onMoveListener && this._onMoveListener.remove();
 
-    if (!this._mouseDownData) return;
+    if (!this._initialEventData) return;
 
     var inRoot = !this.container || contains(this.container(), e.target);
     var bounds = this._selectRect;
-    var click = this.isClick(e.pageX, e.pageY);
+    var click = this.isClick(pageX, pageY);
 
-    this._mouseDownData = null
+    this._initialEventData = null
 
     if(click && !inRoot) {
       return this.emit('reset')
     }
 
     if(click && inRoot)
       return this.emit('click', {
-        x: e.pageX,
-        y: e.pageY,
-        clientX: e.clientX,
-        clientY: e.clientY,
+        x: pageX,
+        y: pageY,
+        clientX: clientX,
+        clientY: clientY,
       })
 
     // User drag-clicked in the Selectable area
     if(!click)
       return this.emit('select', bounds)
-
-    this.selecting = false;
   }
 
-  _openSelector(e) {
-    var { x, y } = this._mouseDownData;
-    var w = Math.abs(x - e.pageX);
-    var h = Math.abs(y - e.pageY);
+  _handleMoveEvent(e) {
+    var { x, y } = this._initialEventData;
```

Comment:
I see a lot of `var` and `let`. Just being picky, but the vars could be turned into lets since I also see consts in the code base.